graviti/dataframe/frame.py

```diff
@@ -125,8 +125,30 @@ def __getitem__(self, key: Union[str, Iterable[str]]) -> Container:
 
         raise KeyError(key)
 
-    def __setitem__(self, key: str, value: Iterable[Any]) -> None:
-        pass
+    def __setitem__(self, key: str, value: Any) -> None:
+        # self.schema.to_builtin always returns record.
+        if isinstance(value, Container):
+            field = value.schema.copy()
+            column = value.copy()
+        else:
+            if key in self._columns:
+                field = self[key].schema
+                column = field.container._from_pyarrow(self._pylist_to_pyarrow(value, field), field)
+            else:
+                array = pa.array(value)
+                field = pt.PortexType.from_pyarrow(array.type)
+                column = field.container._from_pyarrow(array, field)
+
+        if len(column) != len(self):
```

Comment:
can we check the length earlier?

Reply:
We cannot know the length of iterable value。